subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -144,17 +144,15 @@ static const struct bt_mesh_health_srv_cb health_srv_cb = {
 };
 #endif /* CONFIG_BT_MESH_SHELL_HEALTH_SRV_INSTANCE */
 
-#ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
```

Comment:
I guess conditional compilation should be moved back.

Reply:
I don't think it is needed, as from the Bluetooth protocol side, if LCD Server model is not present on the device, the provisioner would never know whether metadata exist or not.

Reply:
In this aspect you are absolutely right. However, I look at this from a different angle. Is compiler able to understand that whether the global variable `health_srv_meta` is used or isn't to optimize it.
If we add conditional compilation, we will be 100% sure it is removed from the final image.

Reply:
Hm. I do not think this is solved. Mark as unresolved to fix it.

Reply:
@omkar3141, how about this?

Reply:
The problem is, on one side reviewers are arguing to remove unnecessary conditional compilation and here, suggestion is opposite.

Reply:
I do not see any contradictions with other comments. Removing conditional compilation in all places where it was pointed was done on purpose with explanation. Isn't any explanation unclear? Please let me know. I'll try to clarify better.